features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/prerequisite/CpuRequirementChecker.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.prerequisite
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.internal.persistence.file.listFilesSafe
+import com.datadog.android.sessionreplay.SystemRequirementConfiguration
+import java.io.File
+
+internal class CpuRequirementChecker(
+    private val cpuDirFile: File = File(DIR_PATH),
+    private val internalLogger: InternalLogger
+) : SystemRequirementChecker {
+
+    private var checkedValue: Int? = null
+
+    override fun checkMinimumRequirement(systemRequirementConfiguration: SystemRequirementConfiguration): Boolean {
+        return readCpuCoreNumber() >= systemRequirementConfiguration.minCPUCoreNumber
+    }
+
+    override fun name(): String = CPU_CHECK_NAME
+
+    override fun checkedValue(): Any? = checkedValue
+
+    private fun readCpuCoreNumber(): Int {
+        val files = cpuDirFile.listFilesSafe(internalLogger) { _, name -> name.matches(REGEX_CPU_CORE_FILE) }
+        val value = files?.size ?: fallbackReadCpuCoreNumber()
+        checkedValue = value
+        return value
+    }
+
+    private fun fallbackReadCpuCoreNumber(): Int {
+        return Runtime.getRuntime().availableProcessors()
```

Comment:
why we cannot rely solely on this and need to read `/sys/devices/system/cpu/` as a primary option?

Reply:
In its doc it says:
> Returns the number of processors available to the Java virtual machine.


so that might not be the physical number of cores  in some cases I think